pkg/package/manifest.go

```diff
@@ -77,6 +77,28 @@ func LoadManifest(_fs fs.FS, path string) (Manifest, error) {
 	return manifest, err
 }
 
+func (m *Manifest) ApplyDefaultVars(vars VarSet) error {
+	for k, schema := range m.VariableSchemas {
+		var parsedValue interface{}
+
+		if _, ok := vars[k]; !schema.ReadOnly && !ok {
+			if schema.Default != "" {
+				vars[k] = schema.Default
+			}
+		}
+
+		if err := json.Unmarshal([]byte(vars[k]), &parsedValue); err != nil {
+			parsedValue = vars[k]
+		}
+
+		if err := schema.Validate(parsedValue); err != nil && parsedValue != "" {
```

Comment:
we don't need to validate it at runtime, this could be added to the `corral package validate`.